core/trino-main/src/main/java/io/trino/sql/planner/plan/AggregationNode.java

```diff
@@ -100,12 +108,34 @@ public AggregationNode(
         checkArgument(preGroupedSymbols.isEmpty() || groupingSets.getGroupingKeys().containsAll(preGroupedSymbols), "Pre-grouped symbols must be a subset of the grouping keys");
         this.preGroupedSymbols = ImmutableList.copyOf(preGroupedSymbols);
 
-        ImmutableList.Builder<Symbol> outputs = ImmutableList.builder();
-        outputs.addAll(groupingSets.getGroupingKeys());
-        hashSymbol.ifPresent(outputs::add);
-        outputs.addAll(aggregations.keySet());
+        ImmutableList.Builder<Symbol> outputsBuilder = ImmutableList.builder();
+        outputsBuilder.addAll(groupingSets.getGroupingKeys());
+        hashSymbol.ifPresent(outputsBuilder::add);
+        outputsBuilder.addAll(aggregations.keySet());
+        if (step.isOutputPartial() && rawInputMaskSymbol.isPresent()) {
+            // add mask channels
+            aggregations.values().stream()
+                    .map(Aggregation::getMask)
```

Comment:
Do we need mask actually given given that distinct accumulator factory does not support PA

Reply:
yes, it's used to implement filtered aggregations (`ImplementFilteredAggregations`) + used by `MarkDistinctOperator` to replace `DistinctAccumulator` in some cases (`MultipleDistinctAggregationToMarkDistinct`)